core/src/apps/common/payment_request.py

```diff
@@ -57,15 +59,31 @@ def verify_payment_request_is_supported(
         if not _is_coin_swap(payment_request):
             raise DataError("Only COIN SWAP payment requests are supported.")
 
+    def get_expected_nonce(self) -> bytes | None:
+        from storage.cache_common import APP_COMMON_NONCE
+
+        context.cache_get(APP_COMMON_NONCE)
+
+    def delete_nonce_cache(self) -> None:
+        from storage.cache_common import APP_COMMON_NONCE
+
+        context.cache_delete(APP_COMMON_NONCE)
+
     if __debug__:
 
         def _use_debug_key(self) -> None:
             # nist256p1 public key of m/0h for "all all ... all" seed.
+            # Corresponding private key: b"\x05\x62\x35\xb0\x47\x6f\x05\x7f\x27\x65\x21\x97\x24\xf7\xf1\x80\x7d\x58\x80\x2b\x55\x0e\xd5\xbf\x6f\x73\x05\x0a\xf5\x45\x63\x00"
+            # keeping it here for reference in case tests need to be updated!
             self.PUBLIC_KEY = b"\x03\xd9\xd9\x3f\x89\xc6\x96\x3b\x94\xbb\xd7\xa5\x11\x88\x28\xe4\x4c\x1c\x39\x59\x15\xac\xe8\x48\x88\x71\x7f\x56\x8c\xb0\x19\x74\xc3"
 
         def _use_debug_verification(self) -> None:
             self.verify_payment_request_is_supported = lambda payment_request: None
 
+        def _use_debug_nonce_verification(self) -> None:
+            self.get_expected_nonce = lambda: bytes("DEBUG NONCE", "utf-8")
```

Comment:
Should it also support returning `None` after `delete_nonce_cache()` is called?

Not sure if it's needed for the test, but IIUC it would be similar to storage key/value storage works, right?

Reply:
> Should it also support returning `None` after `delete_nonce_cache()` is called?

Not sure I understand this. `delete_nonce_cache` returns `None` both in prod and debug mode.